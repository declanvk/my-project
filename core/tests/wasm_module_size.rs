use std::{collections::HashMap, fmt::Write};

use common::configure_pikevm_builder;
use rayon::iter::{ParallelBridge, ParallelIterator};
use regex_test::{
    anyhow::{self, Context},
    RegexTest,
};
use wahgex_core::{Builder, PikeVM};

mod common;

#[derive(Debug)]
struct WasmSizeResult {
    group: String,
    size: usize,
    name: String,
}

#[test]
fn wasm_module_size_of() {
    let suite = common::suite().unwrap();

    let all_regexes = suite
        .iter()
        .par_bridge()
        .filter_map(|test| {
            if !test.compiles() {
                return None;
            }

            let reg = match compile(PikeVM::builder(), test, test.regexes())
                .context(format!("compiling regex [{}]", test.full_name()))
                .unwrap()
            {
                CompileOutput::Skip => {
                    return None;
                },
                CompileOutput::Compiled(reg) => reg,
            };

            Some(WasmSizeResult {
                group: test.group().into(),
                size: reg.get_wasm().len(),
                name: test.name().into(),
            })
        })
        .collect::<Vec<_>>();

    let mut groups: HashMap<_, Vec<WasmSizeResult>> = HashMap::new();

    for result in all_regexes {
        groups.entry(result.group.clone()).or_default().push(result);
    }

    let formatted = format_grouped_results(groups).unwrap();

    insta::assert_snapshot!(formatted);
}

fn format_grouped_results(
    mut groups: HashMap<String, Vec<WasmSizeResult>>,
) -> anyhow::Result<String> {
    let mut formatted = String::new();
    let mut group_names: Vec<_> = groups.keys().cloned().collect();
    group_names.sort();

    for group in group_names {
        let mut results = groups.remove(&group).unwrap();
        writeln!(&mut formatted, "[{group}]")?;
        results.sort_by(|a, b| a.name.cmp(&b.name));
        let max_name_len = results.iter().map(|res| res.name.len()).max().unwrap_or(0);
        for res in results {
            writeln!(
                &mut formatted,
                "{name:<width$} = {size}",
                name = res.name,
                width = max_name_len,
                size = res.size
            )?;
        }
        writeln!(&mut formatted)?;
    }
    Ok(formatted)
}

#[derive(Debug)]
enum CompileOutput {
    Skip,
    Compiled(PikeVM),
}

fn compile(
    mut builder: Builder,
    test: &RegexTest,
    regexes: &[String],
) -> anyhow::Result<CompileOutput> {
    if !configure_pikevm_builder(test, &mut builder) {
        return Ok(CompileOutput::Skip);
    }

    let re = match builder.build_many(regexes) {
        Ok(re) => re,
        Err(err) if err.is_unsupported() => {
            return Ok(CompileOutput::Skip);
        },
        Err(err) => {
            return Err(err.into());
        },
    };

    Ok(CompileOutput::Compiled(re))
}
