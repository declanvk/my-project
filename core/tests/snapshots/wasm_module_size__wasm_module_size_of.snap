---
source: core/tests/wasm_module_size.rs
expression: formatted
---
[anchored]
greedy                                    = 4653
greedy-earliest                           = 4653
no-match-at-start                         = 5720
no-match-at-start-bounds                  = 5720
no-match-at-start-reverse-anchored        = 6256
no-match-at-start-reverse-anchored-bounds = 6256
no-match-at-start-reverse-inner           = 5959
no-match-at-start-reverse-inner-bounds    = 5959
nongreedy                                 = 4653
nongreedy-all                             = 4653

[bytes]
case-class-ascii        = 3980
case-class-simple-ascii = 3980
case-class-unicode      = 4698
case-one-ascii          = 3788
case-one-unicode        = 3788
dotstar-prefix-ascii    = 3582
dotstar-prefix-unicode  = 3582
invalid-utf8-anchor-100 = 4194
invalid-utf8-anchor-200 = 13566
invalid-utf8-anchor-300 = 7538
mixed-dot               = 7082
negate-ascii            = 3788
negate-unicode          = 5481
null-bytes              = 4381
perl-decimal-ascii      = 3772
perl-decimal-unicode    = 12208
perl-whitespace-ascii   = 3980
perl-whitespace-unicode = 6286
perl-word-ascii         = 3986
perl-word-unicode       = 139737

[crazy]
date1                    = 8568
date2                    = 8568
date3                    = 8568
email-big                = 9235
empty1                   = 3502
empty10                  = 3955
empty11                  = 3955
empty2                   = 3502
empty3                   = 3502
empty4                   = 3700
empty5                   = 3502
empty6                   = 3700
empty7                   = 3502
empty8                   = 3955
empty9                   = 3955
float1                   = 5359
float2                   = 5359
float3                   = 5359
float4                   = 5359
float5                   = 5514
greedy-many-many         = 6356
greedy-many-optional     = 6356
greedy-one-many-many     = 6142
greedy-one-many-optional = 6142
greedy-range-many        = 8513
greedy-range-min-many    = 6142
lazy-many-many           = 6356
lazy-many-optional       = 6356
lazy-one-many-many       = 6142
lazy-one-many-optional   = 6142
lazy-range-many          = 8513
lazy-range-min-many      = 6142
neg-class-ascii          = 5484
neg-class-comma          = 5481
neg-class-comma-space    = 5487
neg-class-letter         = 5484
neg-class-letter-comma   = 5484
neg-class-letter-space   = 5487
neg-class-space          = 5484
neg-class-space-comma    = 5487
nothing-empty            = 2409
nothing-something        = 2409
start-end-empty          = 3689
start-end-empty-many-1   = 4526
start-end-empty-many-2   = 4526
start-end-empty-rep      = 4416
start-end-empty-rep-rev  = 4416
start-end-empty-rev      = 3689

[crlf]
basic                           = 4605
dot-no-crlf                     = 5484
end-no-split                    = 3928
end-no-split-adjacent           = 3928
end-no-split-adjacent-cr        = 3928
end-no-split-adjacent-lf        = 3928
onepass-wrong-crlf-anchored     = 5854
onepass-wrong-crlf-with-capture = 6250
start-end-before-after          = 4303
start-end-empty                 = 4303
start-end-non-empty             = 4303
start-no-split                  = 3935
start-no-split-adjacent         = 3935
start-no-split-adjacent-cr      = 3935
start-no-split-adjacent-lf      = 3935

[earliest]
is-ungreedy           = 3774
look-end-test         = 4990
look-start-test       = 4525
no-greedy-100         = 3774
no-greedy-200         = 4248
no-leftmost-first-100 = 4248
no-leftmost-first-200 = 4248

[empty]
100 = 3955
110 = 3955
120 = 3955
130 = 3955
200 = 3700
210 = 3700
220 = 3955
230 = 3955
240 = 3955
300 = 3955
310 = 3955
320 = 3700
330 = 3955
400 = 3582
500 = 3502
510 = 3502
520 = 3502
600 = 4388
610 = 4171

[expensive]
backtrack-blow-visited-capacity          = 6347274
regression-many-repeat-no-stack-overflow = 6302036

[flags]
1  = 4675
10 = 3774
11 = 5198
2  = 4262
3  = 4262
4  = 5923
5  = 8309
6  = 8309
7  = 8503
8  = 3774
9  = 3774

[fowler/basic]
basic10  = 3582
basic100 = 4480
basic101 = 4480
basic102 = 4480
basic103 = 4480
basic104 = 4480
basic105 = 4242
basic106 = 4247
basic107 = 3952
basic108 = 4351
basic109 = 3394
basic110 = 3855
basic111 = 5959
basic112 = 5959
basic113 = 6147
basic114 = 4056
basic115 = 4056
basic116 = 3819
basic117 = 4025
basic118 = 4025
basic119 = 3819
basic12  = 3582
basic120 = 4056
basic121 = 5959
basic122 = 5962
basic123 = 5962
basic124 = 4262
basic125 = 4262
basic126 = 4054
basic127 = 4240
basic128 = 4240
basic129 = 4351
basic130 = 4799
basic131 = 4428
basic132 = 3955
basic133 = 4870
basic134 = 4640
basic135 = 5417
basic136 = 5067
basic137 = 4886
basic138 = 5013
basic139 = 5856
basic140 = 4540
basic141 = 3955
basic142 = 4426
basic143 = 4905
basic144 = 3582
basic145 = 4191
basic146 = 5985
basic147 = 5197
basic148 = 4011
basic149 = 4011
basic15  = 3478
basic150 = 4874
basic151 = 4056
basic152 = 4629
basic153 = 5565
basic154 = 4877
basic155 = 5493
basic156 = 5457
basic157 = 5445
basic158 = 5436
basic159 = 5075
basic16  = 3819
basic160 = 5798
basic161 = 4432
basic162 = 8767
basic163 = 9550
basic164 = 9550
basic165 = 9550
basic166 = 7446
basic167 = 6681
basic168 = 9085
basic169 = 4294
basic17  = 3819
basic170 = 4656
basic171 = 4240
basic172 = 14231
basic173 = 14231
basic174 = 14231
basic175 = 14231
basic176 = 14231
basic177 = 14231
basic178 = 14231
basic179 = 14231
basic18  = 3877
basic180 = 14231
basic181 = 14231
basic182 = 14231
basic183 = 14231
basic184 = 14231
basic185 = 14231
basic186 = 14231
basic187 = 14231
basic188 = 14231
basic189 = 5029
basic19  = 3817
basic190 = 5893
basic191 = 6309
basic192 = 14983
basic193 = 9450
basic194 = 9450
basic195 = 9450
basic196 = 11813
basic197 = 7114
basic198 = 7114
basic199 = 7114
basic20  = 3689
basic200 = 6720
basic201 = 6720
basic202 = 6720
basic203 = 6720
basic204 = 6720
basic205 = 6720
basic206 = 6068
basic207 = 6068
basic208 = 6068
basic209 = 18419
basic21  = 3689
basic210 = 18169
basic211 = 18169
basic212 = 18169
basic213 = 18169
basic214 = 18419
basic215 = 18419
basic216 = 18419
basic217 = 18419
basic218 = 9388
basic219 = 9390
basic22  = 4279
basic220 = 4294
basic221 = 9382
basic222 = 9384
basic223 = 4288
basic23  = 4833
basic24  = 15512
basic25  = 15512
basic26  = 5453
basic27  = 5365
basic28  = 3582
basic29  = 6698
basic3   = 6263
basic30  = 6698
basic32  = 4820
basic33  = 5277
basic34  = 7275
basic35  = 6345
basic36  = 6727
basic37  = 5257
basic38  = 5257
basic39  = 6217
basic4   = 10253
basic40  = 6217
basic41  = 13824
basic42  = 5571
basic43  = 5571
basic44  = 4486
basic45  = 5182
basic46  = 4737
basic47  = 7073
basic48  = 21266
basic49  = 4011
basic5   = 4770
basic50  = 4011
basic51  = 6148
basic52  = 5481
basic53  = 4161
basic54  = 4161
basic55  = 6389
basic56  = 6389
basic57  = 3582
basic58  = 3774
basic59  = 3774
basic6   = 3580
basic65  = 3580
basic66  = 3580
basic67  = 5481
basic68  = 3817
basic69  = 5175
basic7   = 3819
basic70  = 4056
basic72  = 13681
basic74  = 13681
basic76  = 13681
basic78  = 4351
basic80  = 4786
basic81  = 5516
basic82  = 6348
basic83  = 6348
basic84  = 8375
basic86  = 5856
basic87  = 14572
basic89  = 5952
basic9   = 3582
basic90  = 3394
basic91  = 3855
basic92  = 3689
basic93  = 3773
basic94  = 4056
basic95  = 4056
basic96  = 4056
basic97  = 4242
basic98  = 4480
basic99  = 4480

[fowler/nullsubexpr]
nullsubexpr10 = 4640
nullsubexpr11 = 4568
nullsubexpr12 = 4568
nullsubexpr13 = 4568
nullsubexpr14 = 4568
nullsubexpr15 = 4387
nullsubexpr16 = 4387
nullsubexpr17 = 4387
nullsubexpr18 = 4387
nullsubexpr20 = 4870
nullsubexpr21 = 4870
nullsubexpr22 = 4870
nullsubexpr23 = 4870
nullsubexpr24 = 4640
nullsubexpr25 = 4640
nullsubexpr26 = 4640
nullsubexpr27 = 4640
nullsubexpr28 = 6778
nullsubexpr29 = 6778
nullsubexpr3  = 4870
nullsubexpr30 = 6778
nullsubexpr31 = 6778
nullsubexpr32 = 4870
nullsubexpr33 = 4870
nullsubexpr34 = 4870
nullsubexpr35 = 4870
nullsubexpr36 = 4870
nullsubexpr37 = 4870
nullsubexpr38 = 4870
nullsubexpr39 = 6778
nullsubexpr4  = 4870
nullsubexpr40 = 6778
nullsubexpr41 = 6778
nullsubexpr42 = 6778
nullsubexpr43 = 6778
nullsubexpr46 = 5527
nullsubexpr5  = 4870
nullsubexpr6  = 4870
nullsubexpr64 = 5334
nullsubexpr65 = 5334
nullsubexpr66 = 5334
nullsubexpr68 = 5104
nullsubexpr69 = 5104
nullsubexpr7  = 4640
nullsubexpr70 = 5104
nullsubexpr72 = 5792
nullsubexpr73 = 5792
nullsubexpr74 = 5792
nullsubexpr8  = 4640
nullsubexpr9  = 4640

[fowler/repetition]
repetition-expensive100 = 31136
repetition-expensive102 = 30680
repetition-expensive104 = 30237
repetition-expensive106 = 29807
repetition-expensive108 = 29409
repetition-expensive110 = 29026
repetition-expensive112 = 28652
repetition-expensive114 = 28277
repetition-expensive115 = 27916
repetition-expensive127 = 6455
repetition-expensive129 = 6226
repetition-expensive130 = 7964
repetition-expensive131 = 9703
repetition-expensive132 = 11440
repetition-expensive134 = 25119
repetition-expensive136 = 24647
repetition-expensive137 = 24316
repetition-expensive138 = 23985
repetition-expensive139 = 23655
repetition-expensive141 = 6455
repetition-expensive143 = 6226
repetition-expensive149 = 6455
repetition-expensive151 = 6226
repetition-expensive153 = 7964
repetition-expensive155 = 9703
repetition-expensive156 = 11440
repetition-expensive158 = 25119
repetition-expensive160 = 24647
repetition-expensive162 = 24316
repetition-expensive164 = 23985
repetition-expensive165 = 23655
repetition-expensive167 = 6455
repetition-expensive169 = 6226
repetition-expensive90  = 6997
repetition-expensive91  = 6785
repetition-expensive92  = 9596
repetition-expensive93  = 12482
repetition-expensive94  = 15440
repetition-expensive95  = 18473
repetition-expensive96  = 21672
repetition-expensive97  = 24960
repetition-expensive98  = 28326
repetition10            = 11182
repetition11            = 19063
repetition12            = 27018
repetition14            = 11182
repetition15            = 19063
repetition16            = 27018
repetition18            = 11713
repetition20            = 11182
repetition21            = 19063
repetition22            = 27018
repetition24            = 11182
repetition25            = 19063
repetition26            = 27018
repetition28            = 11713
repetition30            = 11182
repetition31            = 19063
repetition32            = 27018
repetition34            = 11182
repetition35            = 19063
repetition36            = 27018
repetition38            = 11713
repetition40            = 11182
repetition41            = 19063
repetition42            = 27018
repetition44            = 11182
repetition46            = 19063
repetition47            = 27018
repetition50            = 11713
repetition52            = 11182
repetition53            = 19063
repetition54            = 27018
repetition56            = 11182
repetition57            = 19063
repetition59            = 27018
repetition61            = 11713
repetition63            = 11182
repetition64            = 19063
repetition65            = 27018
repetition67            = 11182
repetition68            = 19063
repetition70            = 27018
repetition73            = 11713
repetition75            = 11182
repetition76            = 19063
repetition77            = 27018
repetition79            = 11182
repetition80            = 19063
repetition81            = 27018
repetition83            = 11713

[iter]
1                            = 3582
2                            = 3582
anchored1                    = 3582
anchored2                    = 3582
anchored3                    = 3582
empty1                       = 3502
empty10                      = 3955
empty11                      = 3955
empty2                       = 3502
empty3                       = 3502
empty4                       = 3700
empty5                       = 3502
empty6                       = 3700
empty7                       = 3502
empty8                       = 3955
empty9                       = 3955
nonempty-followedby-empty    = 6804
nonempty-followedby-oneempty = 6804
nonempty-followedby-onemixed = 6804
nonempty-followedby-twomixed = 6804
start1                       = 3478
start2                       = 3478

[leftmost-all]
alt    = 4964
dotall = 5478
multi  = 6220

[line-terminator]
carriage                         = 4155
dot-changes-with-line-terminator = 5478
non-ascii                        = 4505
non-word-byte                    = 4503
not-line-feed                    = 4503
nul                              = 4503
word-byte                        = 4505

[misc]
anchor-start-end-line      = 4755
ascii-literal              = 3582
ascii-literal-anchored     = 3582
ascii-literal-anchored-not = 3582
ascii-literal-not          = 3582
one-literal-edge           = 4056
prefix-literal-match       = 3952
prefix-literal-match-ascii = 3952
prefix-literal-no-match    = 3952
suffix-100                 = 6650
suffix-200                 = 6844
suffix-300                 = 6844
suffix-400                 = 6844
suffix-500                 = 7083
suffix-600                 = 7083
terminates                 = 3877

[multiline]
basic1                    = 4503
basic1-crlf               = 4605
basic1-crlf-cr            = 4605
basic2                    = 4201
basic2-crlf               = 4303
basic2-crlf-cr            = 4303
basic3                    = 3883
basic3-crlf               = 3935
basic3-crlf-cr            = 3935
basic4                    = 3878
basic4-crlf               = 3928
basic4-crlf-cr            = 3928
basic5                    = 3963
basic5-crlf               = 4015
basic5-crlf-cr            = 4015
basic6                    = 3905
basic6-crlf               = 3957
basic6-crlf-cr            = 3957
basic7                    = 3900
basic7-crlf               = 3950
basic7-crlf-cr            = 3950
basic8                    = 3958
basic8-crlf               = 4008
basic8-crlf-cr            = 4008
basic9                    = 4201
basic9-crlf               = 4303
match-line-100            = 6408
match-line-100-crlf       = 6514
match-line-100-crlf-cr    = 6514
match-line-200            = 4709
match-line-200-crlf       = 4814
match-line-200-crlf-cr    = 4814
repeat1                   = 4465
repeat1-crlf              = 4567
repeat1-crlf-cr           = 4567
repeat1-no-multi          = 4416
repeat1-no-multi-crlf     = 4416
repeat1-no-multi-crlf-cr  = 4416
repeat10                  = 5210
repeat10-crlf             = 5260
repeat10-crlf-cr          = 5260
repeat10-no-multi         = 5187
repeat10-no-multi-crlf    = 5187
repeat10-no-multi-crlf-cr = 5187
repeat11                  = 4146
repeat11-crlf             = 4198
repeat11-crlf-cr          = 4198
repeat11-no-multi         = 4120
repeat11-no-multi-crlf    = 4120
repeat11-no-multi-crlf-cr = 4120
repeat12                  = 3883
repeat12-crlf             = 3935
repeat12-crlf-cr          = 3935
repeat12-no-multi         = 3394
repeat12-no-multi-crlf    = 3394
repeat12-no-multi-crlf-cr = 3394
repeat13                  = 4141
repeat13-crlf             = 4191
repeat13-crlf-cr          = 4191
repeat13-no-multi         = 4118
repeat13-no-multi-crlf    = 4118
repeat13-no-multi-crlf-cr = 4118
repeat14                  = 3878
repeat14-crlf             = 3928
repeat14-crlf-cr          = 3928
repeat14-no-multi         = 3855
repeat14-no-multi-crlf    = 3855
repeat14-no-multi-crlf-cr = 3855
repeat15                  = 4179
repeat15-crlf             = 4229
repeat15-crlf-cr          = 4229
repeat15-no-multi         = 4156
repeat15-no-multi-crlf    = 4156
repeat15-no-multi-crlf-cr = 4156
repeat16                  = 4360
repeat16-crlf             = 4410
repeat16-crlf-cr          = 4410
repeat16-no-multi         = 4337
repeat16-no-multi-crlf    = 4337
repeat16-no-multi-crlf-cr = 4337
repeat17                  = 4503
repeat17-crlf             = 4605
repeat17-crlf-cr          = 4605
repeat17-no-multi         = 4454
repeat17-no-multi-crlf    = 4454
repeat17-no-multi-crlf-cr = 4454
repeat18                  = 4522
repeat18-crlf             = 4624
repeat18-crlf-cr          = 4624
repeat18-no-multi         = 4473
repeat18-no-multi-crlf    = 4473
repeat18-no-multi-crlf-cr = 4473
repeat2                   = 4605
repeat2-crlf              = 4657
repeat2-crlf-cr           = 4657
repeat2-no-multi          = 4579
repeat2-no-multi-crlf     = 4579
repeat2-no-multi-crlf-cr  = 4579
repeat3                   = 4880
repeat3-crlf              = 4932
repeat3-crlf-cr           = 4932
repeat3-no-multi          = 4854
repeat3-no-multi-crlf     = 4854
repeat3-no-multi-crlf-cr  = 4854
repeat4                   = 4558
repeat4-crlf              = 4610
repeat4-crlf-cr           = 4610
repeat4-no-multi          = 4532
repeat4-no-multi-crlf     = 4532
repeat4-no-multi-crlf-cr  = 4532
repeat5                   = 4612
repeat5-crlf              = 4664
repeat5-crlf-cr           = 4664
repeat5-no-multi          = 4586
repeat5-no-multi-crlf     = 4586
repeat5-no-multi-crlf-cr  = 4586
repeat6                   = 4186
repeat6-crlf              = 4238
repeat6-crlf-cr           = 4238
repeat6-no-multi          = 3701
repeat6-no-multi-crlf     = 3701
repeat6-no-multi-crlf-cr  = 3701
repeat7                   = 5136
repeat7-crlf              = 5188
repeat7-crlf-cr           = 5188
repeat7-no-multi          = 4645
repeat7-no-multi-crlf     = 4645
repeat7-no-multi-crlf-cr  = 4645
repeat8                   = 5317
repeat8-crlf              = 5369
repeat8-crlf-cr           = 5369
repeat8-no-multi          = 5291
repeat8-no-multi-crlf     = 5291
repeat8-no-multi-crlf-cr  = 5291
repeat9                   = 5029
repeat9-crlf              = 5079
repeat9-crlf-cr           = 5079
repeat9-no-multi          = 5006
repeat9-no-multi-crlf     = 5006
repeat9-no-multi-crlf-cr  = 5006

[no-unicode]
anchored-iter-empty-utf8          = 3502
case1                             = 3788
case2                             = 3980
case3                             = 4698
case4                             = 3980
decimal-ascii                     = 3772
decimal-unicode                   = 12208
dotstar-prefix1                   = 3582
dotstar-prefix2                   = 3582
invalid-utf8-literal1             = 3582
iter1-bytes                       = 3357
iter1-utf8                        = 3502
iter2-bytes                       = 3357
mixed                             = 6311
negate1                           = 5481
negate2                           = 3788
null-bytes1                       = 4002
space-ascii                       = 3980
space-unicode                     = 6286
unanchored-invalid-utf8-match-100 = 3582
unanchored-invalid-utf8-nomatch   = 5807
word-ascii                        = 3986
word-unicode                      = 139737

[overlapping]
alt-all-100                             = 4248
alt-leftmost-first-100                  = 4248
empty-000                               = 3502
empty-alt-000                           = 3955
empty-alt-010                           = 3955
greedy-dotstar-matches-everything-100   = 5856
iter1-bytes                             = 3357
iter1-incomplete-utf8                   = 3357
iter1-utf8                              = 3502
repetition-plus-all-100                 = 3774
repetition-plus-all-110                 = 4248
repetition-plus-all-200                 = 4653
repetition-plus-leftmost-first-100      = 3774
repetition-plus-leftmost-first-110      = 4248
repetition-plus-leftmost-first-200      = 4653
repetition-star-all-100                 = 3955
repetition-star-all-200                 = 4834
repetition-star-leftmost-first-100      = 3955
repetition-star-leftmost-first-200      = 4834
scratch                                 = 6459
start-end-rep-all                       = 4835
start-end-rep-leftmost-first            = 4835
ungreedy-dotstar-matches-everything-100 = 5856

[regression]
adjacent-line-boundary-100                     = 6408
adjacent-line-boundary-200                     = 6408
aho-corasick-100                               = 5761
alt-in-alt-100                                 = 4767
alt-in-alt-200                                 = 7202
anchored-prefix-100                            = 5617
anchored-prefix-200                            = 5617
anchored-prefix-300                            = 3713
ascii-word-underscore                          = 3794
captures-after-dfa-premature-end-100           = 5938
captures-after-dfa-premature-end-200           = 6048
captures-after-dfa-premature-end-300           = 4874
captures-repeat                                = 5175
captures-wrong-order                           = 3960
empty-flag-expr                                = 3502
empty-group-with-unicode                       = 4290
flags-are-unset                                = 5602
fowler-basic154-unanchored                     = 4877
hir-optimization-out-of-order-class            = 4203
i429-10                                        = 5907
i429-4                                         = 6189
i429-5                                         = 3890
i429-6                                         = 5182
i969                                           = 6445
impossible-branch                              = 6396
improper-reverse-suffix-optimization           = 8063
interior-anchor-capture                        = 4698
leftmost-first-prefix                          = 4504
literal-panic                                  = 12969
lits-unambiguous-100                           = 5218
lits-unambiguous-200                           = 10107
many-alternates                                = 4715
missed-match                                   = 11156
negated-char-class-100                         = 5484
negated-char-class-200                         = 5484
negated-full-byte-range                        = 3345
partial-anchor                                 = 4415
partial-anchor-alternate-begin                 = 4415
partial-anchor-alternate-end                   = 4324
prefilter-with-aho-corasick-standard-semantics = 4869
regex-to-glob                                  = 5625
reverse-inner-plus-shorter-than-expected       = 49872
reverse-inner-short                            = 6817
reverse-suffix-100                             = 4758
reverse-suffix-200                             = 4758
ruff-whitespace-around-keywords                = 4641
strange-anchor-non-complete-prefix             = 3879
strange-anchor-non-complete-suffix             = 3935
unicode-case-lower-nocase-flag                 = 33601
unsorted-binary-search-100                     = 3983
unsorted-binary-search-200                     = 3983
y-or-endl                                      = 4566
zero-or-end                                    = 4335

[set]
basic10                = 4545
basic10-leftmost-first = 4545
basic100               = 6846
basic101               = 6846
basic102               = 6846
basic120               = 3582
basic121               = 5932
basic122               = 7155
basic130               = 4785
basic20                = 4545
basic30                = 4545
basic40                = 4545
basic50                = 4545
basic60                = 5263
basic60-leftmost-first = 5263
basic61                = 5263
basic61-leftmost-first = 5263
basic70                = 7533
basic71                = 7533
basic80                = 6188
basic81                = 6188
basic82                = 6188
basic90                = 5544
basic91                = 5544
caps-010               = 9418
caps-020               = 9884
caps-030               = 10636
caps-110               = 8640
caps-120               = 8640
caps-121               = 8640
empty10                = 4469
empty10-leftmost-first = 4469
empty11                = 4469
empty11-leftmost-first = 4469
empty20                = 4469
empty20-leftmost-first = 4469
empty21                = 4469
empty21-leftmost-first = 4469
empty22                = 4469
empty23                = 4469
empty30                = 4469
empty30-leftmost-first = 4469
empty31                = 4469
empty31-leftmost-first = 4469
empty40                = 4545
empty40-leftmost-first = 4545
nomatch10              = 4545
nomatch20              = 6188
nomatch30              = 2409
nomatch40              = 7353

[unicode]
class-emoji1          = 23942
class-emoji2          = 23942
class-emoji3          = 19942
class-emoji4          = 19942
class-gcb1            = 8778
class-gcb2            = 4294
class-gcb3            = 4294
class-gcb4            = 4294
class-gcb5            = 9699
class-gcb6            = 4056
class-gencat1         = 31999
class-gencat10        = 12829
class-gencat11        = 5375
class-gencat12        = 126244
class-gencat13        = 9316
class-gencat14        = 4056
class-gencat15        = 37824
class-gencat16        = 68188
class-gencat17        = 22744
class-gencat18        = 23809
class-gencat19        = 11464
class-gencat2         = 12588
class-gencat20        = 70586
class-gencat21        = 33798
class-gencat22        = 13045
class-gencat23        = 127976
class-gencat24        = 104076
class-gencat25        = 26100
class-gencat26        = 42602
class-gencat27        = 46976
class-gencat28        = 4056
class-gencat29        = 6082
class-gencat3         = 5841
class-gencat30        = 45754
class-gencat31        = 5882
class-gencat32        = 5879
class-gencat33        = 37074
class-gencat34        = 55630
class-gencat35        = 5575
class-gencat36        = 126051
class-gencat37        = 33052
class-gencat4         = 4028
class-gencat5         = 11497
class-gencat6         = 10429
class-gencat7         = 12014
class-gencat8         = 5669
class-gencat9         = 5372
class-sentence-break1 = 38991
class-sentence-break2 = 38991
class-sentence-break3 = 13670
class-sentence-break4 = 13670
class-sentence-break5 = 8452
class-word-break1     = 5575
class-word-break2     = 5575
class-word-break3     = 6288
class-word-break4     = 5635
class-word-break5     = 14568
class1                = 4692
class10               = 38023
class2                = 33798
class3                = 33997
class4                = 43209
class5                = 43209
class6                = 33997
class7                = 33251
class8                = 35268
class9                = 126451
literal1              = 4056
literal2              = 4248
literal3              = 4248
literal4              = 4025
perl1                 = 139737
perl2                 = 139737
perl3                 = 134789
perl4                 = 12208
perl5                 = 12208
perl6                 = 15746
perl7                 = 6286
perl8                 = 6286
perl9                 = 8621

[utf8]
empty-utf8no                                  = 3357
empty-utf8no-anchored                         = 3357
empty-utf8no-anchored-bounds                  = 3357
empty-utf8no-anchored-bounds-overlapping      = 3357
empty-utf8no-anchored-endbound                = 3357
empty-utf8no-anchored-endbound-overlapping    = 3357
empty-utf8no-anchored-overlapping             = 3357
empty-utf8no-anchored-startbound              = 3357
empty-utf8no-anchored-startbound-overlapping  = 3357
empty-utf8no-bounds                           = 3357
empty-utf8no-bounds-overlapping               = 3357
empty-utf8no-overlapping                      = 3357
empty-utf8no-startbound                       = 3357
empty-utf8no-startbound-overlapping           = 3357
empty-utf8yes                                 = 3502
empty-utf8yes-anchored                        = 3502
empty-utf8yes-anchored-bounds                 = 3502
empty-utf8yes-anchored-bounds-overlapping     = 3502
empty-utf8yes-anchored-endbound               = 3502
empty-utf8yes-anchored-endbound-overlapping   = 3502
empty-utf8yes-anchored-overlapping            = 3502
empty-utf8yes-anchored-startbound             = 3502
empty-utf8yes-anchored-startbound-overlapping = 3502
empty-utf8yes-bounds                          = 3502
empty-utf8yes-bounds-overlapping              = 3502
empty-utf8yes-overlapping                     = 3502
empty-utf8yes-startbound                      = 3502
empty-utf8yes-startbound-overlapping          = 3502
